Automation code for discovery work flow.

prasant750h opened the pull request:
This PR introduces a new discovery module that implements automated cluster validation workflows using Molecule for orchestration and pytest for verification. The discovery module follows the same architectural pattern as the existing prepare_oim module, ensuring consistency across the automation framework.

prasant750h committed:
Add discovery validation suite and lint config

prasant750h committed:
Refactor discovery module to Molecule workflow

- Move discovery module to automation_library/discovery/ package structure
- Refactor all modules (checks, omnia_sh, prepare_oim, discovery) into package layout
- Move playbook execution and prechecks from pytest tests to Molecule converge phase
- Simplify discovery tests to verify-only (no playbook execution in tests)
- Remove unused run_discovery_validation.py and bulk validation functions
- Update converge.yml to execute ansible-playbook discovery/discovery.yml in omnia_core
- Remove discovery_validation config from user_config.yml
- Fix all imports and maintain backward compatibility
- Add missing __init__.py files for proper package structure
- Achieve 9.99/10 pylint score for discovery module

Discovery now follows prepare_oim flow:
- Molecule converge: runs prechecks + discovery playbook
- Pytest verify: validates post-discovery cluster state only

balajikumaran-c-s commented on molecule/discovery/converge.yml:
```diff
@@ -0,0 +1,204 @@
+          ║   - Discovery playbook syntax errors
+          ║   - Network connectivity issues to cluster nodes
+          ║   - Missing or invalid cluster configuration
+          ║   - Insufficient permissions on target nodes
```
this error msg is not required. We are not touching the target nodes while running discovery.yml

balajikumaran-c-s commented on molecule/discovery/tests/test_discovery.py:
```diff
@@ -0,0 +1,278 @@
+    assert result.get("success"), result.get("error") or result.get("details") or "Validation failed"
+
+
+def _is_verbose() -> bool:
```
this verbose function is not required

abhishek-sa1 commented on automation_library/discovery/functions/discovery_func.py:
```diff
@@ -0,0 +1,1649 @@
+    return _result("Slurm GPU", ok, details="GPU check succeeded" if ok else None, error=None if ok else f"GPU check failed (rc={rc}): {err or out}", rc=rc, out=out, err=err, node=n)
+
+
+def validate_slurm_ib(host) -> Dict[str, Any]:
```
ensure doc string present for all functions

prasant750h committed:
Address code review feedback: add docstrings, remove unnecessary error messages, and clean up verbose function

- Add comprehensive docstrings to all functions in discovery_func.py (abhishek-sa1)
- Remove unnecessary target node error messages from converge.yml (balajikumaran-c-s)
- Remove unused verbose function from test_discovery.py (balajikumaran-c-s)

balajikumaran-c-s commented on user_config.yml:
```diff
@@ -32,13 +32,13 @@ skip_on_failure: false
 
 # OIM server IP address (REQUIRED)
-oim_server_ip: ""
+oim_server_ip: "182.168.0.71"
```
Remove credentials

balajikumaran-c-s commented on user_config.yml:
```diff
@@ -106,10 +106,10 @@ force_configure_pxe: true
 
 # NFS server IP address
-nfs_server_ip: ""
+nfs_server_ip: "100.96.20.223"
```
remove ip
